compiler/noirc_evaluator/src/ssa/mod.rs

```diff
@@ -126,6 +126,9 @@ pub fn primary_passes(options: &SsaEvaluatorOptions) -> Vec<SsaPass> {
         ),
         // Run mem2reg with the CFG separated into blocks
         SsaPass::new(Ssa::mem2reg, "Mem2Reg"),
+        // Run DIE in case mem2reg left some Load it could not eliminate, but it's also unused
+        // and could cause issues after flattening, such as loading from empty slices.
+        SsaPass::new(Ssa::dead_instruction_elimination, "Dead Instruction Elimination"),
```

Comment:
Won't this have the `flattened` flag set while this occurs before flattening? The only thing that seems to be saving us is that we are also checking whether an ACIR block has a single block, but this check seems very brittle.

Reply:
Thanks, I completely forgot about that. Added a `die_pre_check` that calls the flattening post-check to keep this from happening by accident.